File: hm6/Startup.fs
namespace hm6

open System
open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Hosting
open Microsoft.AspNetCore.Http
open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Hosting
open Giraffe
open hm6.CalculatorHandler


type Startup() =
    
    let webApp =
            choose [route "/ping" >=> text "pong"
                    route "/calculate" >=> calculatorHttpHandler]
    member _.ConfigureServices(services: IServiceCollection) =
            services.AddGiraffe() |> ignore

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
    member _.Configure(app: IApplicationBuilder, env: IWebHostEnvironment) =
            app.UseGiraffe webApp
            app.Use |> ignore

File: HM13/WebApplication/Startup.cs
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Microsoft.Extensions.Logging;
using WebApplication.Controllers;
using WebApplication.DB;
using WebApplication.Models;

namespace WebApplication;

public class Startup
{
    public Startup(IConfiguration configuration) =>
        Configuration = configuration;

    public IConfiguration Configuration { get; }

    public void ConfigureServices(IServiceCollection services)
    {
        services.AddMiniProfiler();
        
        services.AddScoped<ExpressionsCache>();
        services.AddSingleton<ICalculator, Calculator>();
        services.AddSingleton<ICachedCalculator, CachedCalculator>();
        services.AddControllersWithViews();
        services.AddSingleton<ExceptionHandler>();
    }

    public void Configure(IApplicationBuilder app, IWebHostEnvironment env) =>
        (env.IsDevelopment()
            ? app.UseDeveloperExceptionPage()
            : app.UseExceptionHandler("/Home/Error").UseHsts()).UseHttpsRedirection().UseStaticFiles()
        .UseRouting()
        .UseAuthorization()
        .UseMiniProfiler()
        .UseEndpoints(endpoints =>
            endpoints.MapControllerRoute("default", "{controller=Home}/{action=Index}/{id?}"));
}
